[Alerts][Docs] Extended README.md and the user docs with the licensing information. (#92564)
elastic/kibana: x-pack/plugins/alerts/README.md

}
```

## Licensing

> Currently most of the alerts are licensed as "basic". But some alert types are at least "gold" level, eg tracking containment alert.

Reviewer: ```suggestion
Currently most of the alerts are free features. But some alert types are subscription features, such as the tracking containment alert.
```

## Documentation

You should create documentation for the new alert type. Make an entry in the alert type index [`docs/user/alerting/alert-types.asciidoc`](../../../docs/user/alerting/alert-types.asciidoc) that points to a new document for the alert type that should be in the directory [`docs/user/alerting/alert-types`](../../../docs/user/alerting/alert-types).

> 

Reviewer 2: Presumably solution specific alert types should be under their directories rather than `docs/user/alerting/alert-types/`? 🤔 

If so, best to clarify that

Author: I will replace this part with more generic definition. 👍

> You should also create some asciidoc for the new alert type. An entry should be made in the alert type index - [`docs/user/alerting/alert-types.asciidoc`](../../../docs/user/alerting/alert-types.asciidoc) which points to a new document for the alert type that should be in the directory [`docs/user/alerting/alert-types`](../../../docs/user/alerting/alert-types).

Reviewer 3: ```suggestion
You should create documentation for the new alert type. Make an entry in the alert type index [`docs/user/alerting/alert-types.asciidoc`](../../../docs/user/alerting/alert-types.asciidoc) that points to a new document for the alert type that should be in the directory [`docs/user/alerting/alert-types`](../../../docs/user/alerting/alert-types).
```

## Tests

> The alert type should have both jest tests and functional tests.

Reviewer 2: Is it worth clarifying what *should* be tested (such as alert specific action groups, parameters etc.) vs. things that we guarantee as a framework and do not need to be independently tested by each alert type? 🤔
